analyses/long-format-table-utils/annotator/annotator-api.R

```diff
@@ -0,0 +1,358 @@
+                 as.integer(0))) {
+    stop(paste0("analyses/long-format-table-utils/annotator/annotation-data/",
+                "oncokb-cancer-gene-list.tsv ",
+                "has NAs in the `Hugo Symbol` column.\n",
+                "Check data integrity. Submit a data question GitHub issue."))
+  }
+  # assert all symbols are unique
+  if (!identical(nrow(hgsb_oncokb_cgene_oncogene_tsg_df),
+                 length(unique(dplyr::pull(hgsb_oncokb_cgene_oncogene_tsg_df,
+                                           `Hugo Symbol`))))) {
+    stop(paste0("analyses/long-format-table-utils/annotator/annotation-data/",
+                "oncokb-cancer-gene-list.tsv ",
+                "has duplicates in the `Hugo Symbol` column.\n",
+                "Check data integrity. Submit a data question GitHub issue."))
+  }
+
+  # assert no NA in hgsb_gtype_df
+  if (!identical(sum(is.na(hgsb_gtype_df)), as.integer(0))) {
+    stop(paste0("analyses/fusion_filtering/references/genelistreference.txt ",
+                "has NAs in the table.\n",
+                "Check data integrity. Submit a data question GitHub issue."))
+  }
+
+  # assert all ensg_ids and gene_symbols are not NA
+  if (!identical(sum(is.na(ensg_rmtl_df$ensg_id)), as.integer(0))) {
+    stop(paste0("ensg-hugo-rmtl-v1-mapping.tsv ",
+                "has NAs in the ensg_id column.\n",
+                "Check data integrity. Submit a data question GitHub issue."))
+  }
+  # assert all ensg_id are unique
+  if (!identical(length(unique(ensg_rmtl_df$ensg_id)), nrow(ensg_rmtl_df))) {
+    stop(paste0("ensg-hugo-rmtl-v1-mapping.tsv ",
+                "has duplicates in the ensg_id column.\n",
+                "Check data integrity. Submit a data question GitHub issue."))
+  }
+  # asert all rmtl NAs have version NAs, vice versa
+  if (!identical(is.na(ensg_rmtl_df$rmtl),
+                 is.na(ensg_rmtl_df$version))) {
+    stop(paste0("ensg-hugo-rmtl-v1-mapping.tsv ",
+                "has rmtl column NAs with version column non-NAs, or",
+                "version column NAs with rmtl column non-NAs\n",
+                "Check data integrity. Submit a data question GitHub issue."))
+  }
+
+  # assert all cancer_groups are not NA
+  if (!identical(sum(is.na(cgroup_efo_mondo_df$cancer_group)), as.integer(0))) {
+    stop(paste0("efo-mondo-map.tsv has NAs in the cancer_group column.\n",
+                "Check data integrity. Submit a data question GitHub issue."))
+  }
+  # assert all cancer_groups are unique.
+  if (!identical(length(unique(cgroup_efo_mondo_df$cancer_group)),
+                 nrow(cgroup_efo_mondo_df))) {
+    stop(
+      paste0("efo-mondo-map.tsv has duplicates in the cancer_group column.\n",
+             "Check data integrity. Submit a data question GitHub issue."))
+  }
+
+
+  # Process annotation data for joining input table
+  #
+  # pp in variable names means preprocessed
+  #
+  # %>% is not used, so the user's environment will not be modified by importing
```

Comment:
This seems counterintuitive to me as `%>%` *is* used, a few lines down. Would you be able to clarify?

Reply:
Thank you for pointing this out. I forgot to remove this comment when refactoring.